clarify that we use GA with ComEX/MPI

orbeckst opened the pull request:
- updated GA section
- added Daily 2014 paper
- fix #32

orbeckst committed:
clarify that we use GA with ComEX/MPI

- updated GA section
- added Daily 2014 paper

orbeckst commented:
@mkhoshle can you please check that it is correct that we use GA with MPI?

mkhoshle commented on applications.tex:
```diff
@@ -54,18 +54,19 @@ \subsection{MPI for Python (\package{mpi4py})}
-Global Arrays in NumPy (GAiN) extends GA to Python through Numpy \cite{GAiN}. 
+The \emph{Global Arrays} (GA) toolkit provides users with a language interface that allows them to distribute data while maintaining the type of global index space and programming syntax similar to what is available when programming on a single processor \cite{GA}. The Global Arrays toolkit allows manipulating physically distributed dense multi-dimensional arrays without explicitly defining commu ...
+Communication is configured by a runtime, which defaults to the \emph{Communication runtime for Extreme Scale} (ComEx) \cite{Daily:2014aa}.
+ComEx uses share memory for intra-node communication and we employed ComEx with MPI for inter-node communication.
```
"Shared memory"

mkhoshle replied:
I do not like this sentence that is added. What is the point? Are we doing this? No, we just employed GA4py and this stuff are already implemented in that. Anybody who is interested in more detail can read their paper. I do not think we are expected to add these details. This is not a computer science paper.

iparask replied:
I think it should be there if it is what the GA version we used uses. I suggest to be written as

```suggestion
ComEx uses shared memory for intra-node communication. Inter-node communication employs ComEx with MPI.
```

Not being a Computer Engineering/Computer Science paper does not mean it should hide important information. Further up in the paper an argument is made that documentation is not sufficient because dependencies are not mentioned correctly. I wouldn't want to hide this kind of information if it is valid for the version of GA we used

orbeckst replied:
The question how GA performs its communication is something that we all have been asking when we saw the rather good results. We need to state that GA still uses MPI somewhere. I agree with your point https://github.com/hpcanalytics/paper-hpc-py-parallel-mdanalysis/issues/32#issuecomment-461602261 , 
> In MPI in order to get good performance sometimes, we need to have some considerations of the computer architecture and memory hierarchy. However, global array makes it easier for people who are not expert in CS (like MD community) but at the same time want to get good performance.

and it seems that GA does make good choices how it uses MPI. We should not leave this implicit, especially as they have a paper where they explain what they do in detail.

mkhoshle replied:
Alright, but ComEx is the communication library. I believe it is wrong to say that Communication is configured by a runtime. I do not understand the meaning of this sentence. @orbeckst What do you mean by this? Can you clarify?

orbeckst replied:
I agree that this is not a good sentence. The type of communication is _determined_ by the runtime environment. Is the following better?

> The underlying communication is determined by a runtime envronment, which defaults to the \emph{Communication runtime for Extreme Scale} (ComEx) \cite{Daily:2014aa}.

Commit:
Update applications.tex

Co-Authored-By: orbeckst <orbeckst@gmail.com>

orbeckst commented:
@mkhoshle "shared" was fixed. 

I do want to keep the note on MPI in here. We are not making a big deal about it but we are also supplying the information that a conscientious reader might want to have.

mkhoshle commented:
> @mkhoshle "shared" was fixed.
> 
> I do want to keep the note on MPI in here. We are not making a big deal about it but we are also supplying the information that a conscientious reader might want to have.

Ok, but then please have a look at my comments in the review.

orbeckst committed:
fixed description of ComEx runtime environment

orbeckst committed:
condensed GA methods section

- simplified title to "Global Arrays toolkit"
- rewrote some sentences so that they differ from sentences in \cite{GA} (the GA paper)
- summarized mentioning of functions that we don't use
- use "global array" or "global arrays" when referring to the actual data structures as opposed
  to the toolkit (consistent with the usage in the GA paper); we need to check the rest of the
  manuscript for consistency, though

orbeckst commented:
@mkhoshle please have a look. Thanks.

mkhoshle commented:
> @mkhoshle please have a look. Thanks.

Thanks, looks good to me.

orbeckst commented:
Thanks for the quick turnaround.
